### PR TITLE
Add documentation for IPA-Ironic build process

### PR DESCRIPTION
**Pull request by Rozzii**

### CODE REVIEW

File: wow/ipa-ironic-build.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+git `sha` hash of the commit referenced by the `HEAD` pointer on the current branch.
+
+2. The docker image tag is saved to `/tmp/vars.sh` on the builder vm to be used during the combined testing of IPA and Ironic.
+
+3. The script checks whether there is a ironic image in [nordix harbor](https://registry.nordix.org/harbor) with the same name and image tag and if not then the build process will continue.
+
+4. In case the build process continues the script will rely on upstream [patch-image.sh](https://github.com/metal3-io/ironic-image/blob/master/patch-image.sh) to patch the [ironic image](https://github.com/metal3-io/ironic-image/blob/master/Dockerfile). First the patch file is constructed then the docker image will be built
+and the path to the patch file will be supplied with the `docker build` command's `--build-arg` option.
+
+5. When the build process succeeds the newly created image will be uploaded to nordix harbor.
```
Comment (fmuyassarov) [inline wow/ipa-ironic-build.md:53]:
```suggestion
5. When the build process succeeds the newly created container image will be uploaded to nordix harbor.
```

File: wow/ipa-ironic-build.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+## IPA build
+The goal of the IPA build process is to create a custom `.tar` archive that contains the linux kernel and the rootfs image that make up an IPA instance. The
+resulting IPA `.tar` archive is uploaded to [nordix artifactory](https://artifactory.nordix.org/)
+
+The IPA build process uses the python tool called [IPA builder](https://github.com/Nordix/ironic-python-agent-builder.git) to build a selected version of
+[IPA](https://github.com/Nordix/ironic-python-agent). The version of the IPA repository tha will be built is specified via the `IPA_REF` abd the `IPA_BRANCH` environment variables. The `IPA_BRANCH`specifies what branch of IPA will be used during the build and the `IPA_REF` specifies the git commit hash that is used to select a specific version of the repository on the branch. By using the Nordix fork of the IPA repository and custom commit hash and branch name it is possible to build versions of ipa that are not part of the [upstream IPA repository](https://opendev.org/openstack/ironic-python-agent).
```
Comment (fmuyassarov) [inline wow/ipa-ironic-build.md:60]:
```suggestion
[IPA](https://github.com/Nordix/ironic-python-agent). The version of the IPA repository tha will be built is specified via the `IPA_REF` and the `IPA_BRANCH` environment variables. The `IPA_BRANCH`specifies what branch of IPA will be used during the build and the `IPA_REF` specifies the git commit hash that is used to select a specific version of the repository on the branch. By using the Nordix fork of the IPA repository and custom commit hash and branch name it is possible to build versions of ipa that are not part of the [upstream IPA repository](https://opendev.org/openstack/ironic-python-agent).
```

**Review by fmuyassarov — CHANGES_REQUESTED** [review]
Looks nice. Thank for adding it. Found to nits. Otherwise lgtm.

File: wow/ipa-ironic-build.md (hunk 1 above)
Comment (fmuyassarov) [reply wow/ipa-ironic-build.md:53]: or OCI

Comment (Rozzii) [reply wow/ipa-ironic-build.md]: fixed.

File: wow/ipa-ironic-build.md (hunk 2 above)
Comment (Rozzii) [reply wow/ipa-ironic-build.md]: fixed.

**Review by fmuyassarov — APPROVED** [review]

**Pull request merged**